pylot/simulation/carla_operator.py

```diff
@@ -360,7 +363,20 @@ def __send_hero_vehicle_data(self, stream: WriteStream,
         forward_speed = velocity_vector.magnitude()
         pose = pylot.utils.Pose(vec_transform, forward_speed, velocity_vector,
                                 timestamp.coordinates[0])
-        stream.send(erdos.Message(timestamp, pose))
+        message_stream = []
+        message_stream_filename = 'message_stream.pkl'
+
+        if os.path.exists(message_stream_filename):
+            with open(message_stream_filename, 'rb') as rfp:
+                message_stream = pickle.load(rfp)
+
+        message = erdos.Message(timestamp, pose)
+        message_stream.append(message)
+
+        with open(message_stream_filename, 'wb') as wfp:
+            pickle.dump(message_stream, wfp)
+
+        stream.send(message)
```

Comment:
Can we move this to `data_gatherer.py` too? Look into how we can get data from the pose stream using `erdos.ExtractStream`.